static_src/components/activity_log.jsx

```diff
@@ -74,7 +76,9 @@ export default class ActivityLog extends React.Component {
   render() {
     let content = <div></div>;
 
-    if (this.state.empty) {
+    if (this.state.empty && this.state.hasErrors) {
+      content = <h5 className="test-none_message">An error occured fetching recent activity</h5>;
```

Comment:
> In the current case of the activity log, an error does occur, but some activity items are still fetched (at least thats what appears to happen in my UI).

Yes, this is definitely the case. Events and logs are fetched separately so, it is possible one succeeds and the other fails (it's usually the logs that fail). I think what I've implemented addresses your concerns about UI clutter. The main use case I am aiming for is when there is no activity to show **and** there was an error. Previously in that case we were showing "No recent activity", which feels really misleading to me when there's actually an error.

I didn't want to get too complicated where one succeeds with data and then we would be showing data + error message so, in that case, we would show the data we have and hide any error (unless you're seeing different and that would be a bug).

I think this solution is actually the cleanest without any significant design changes. Does that make sense?